File: GanAndroidApp/app/src/main/java/com/xattacker/util/IOUtility.java
package com.xattacker.util;

import java.io.ByteArrayOutputStream;
import java.io.InputStream;

public final class IOUtility
{
	private IOUtility()
	{
		
	}
	
	public static void readResponse(InputStream aIn, ByteArrayOutputStream aBos) throws Exception
	{
		if (aIn != null && aBos != null)
		{
		  byte[] temp = new byte[256];
		  int index = -1;
		  
		  do
		  {
			  while ((index = aIn.read(temp)) != -1)
			  {
				  aBos.write(temp, 0, index);
				  
				  if (index < temp.length)
				  {
					  break;
				  }
			  }
			  
		  } while (aBos.size() == 0);
		  
		  temp = null;
		}
	}
}

File: GanAndroidApp/GanClient/src/main/java/com/xattacker/util/IOUtility.kt
package com.xattacker.util

import java.io.ByteArrayOutputStream
import java.io.InputStream

object IOUtility
{
    @Throws(Exception::class)
    fun readResponse(aIn: InputStream?, aBos: ByteArrayOutputStream?)
    {
        if (aIn != null && aBos != null)
        {
            var temp = ByteArray(256)
            var index = -1
            do
            {
                while (aIn.read(temp).also {index = it} != -1)
                {
                    aBos.write(temp, 0, index)
                    if (index < temp.size)
                    {
                        break
                    }
                }
            } while (aBos.size() == 0)
        }
    }
}